### PR TITLE
Added RegexField and RangeField

### PR DESCRIPTION
**Pull request by sattyamjjain**

### CODE REVIEW

File: pydictable/field.py [hunk 1]
```diff
@@ -332,3 +336,53 @@ def validate(self, field_name: str, value):
...
+    def validate_dict(self, field_name: str, v):
+        try:
+            assert re.match(self.regex_string, v), f"{v} for {field_name} should be in proper format"
+        except AssertionError as e:
+            raise AssertionError(str(e))
+
+    def validate(self, field_name: str, v):
+        assert isinstance(v, str)
+
+    def regex(self):
```
Comment (pskd73) [inline pydictable/field.py]: You can get it from self right?

File: pydictable/field.py [hunk 2]
```diff
@@ -332,3 +336,53 @@ def validate(self, field_name: str, value):
...
+
+    def from_dict(self, v):
+        return v
+
+    def to_dict(self, v):
+        return v
+
+    def validate_dict(self, field_name: str, v):
+        try:
+            assert re.match(self.regex_string, v), f"{v} for {field_name} should be in proper format"
```
Comment (pskd73) [inline pydictable/field.py]: assert itself raises AssertionError right? Why to catch and raise again?

File: pydictable/field.py [hunk 3]
```diff
@@ -332,3 +336,53 @@ def validate(self, field_name: str, value):
...
+            raise AssertionError(str(e))
+
+    def validate(self, field_name: str, v):
+        assert isinstance(v, str)
+
+    def regex(self):
+        return self.regex_string
+
+
+class RangeField(Field):
```
Comment (pskd73) [inline pydictable/field.py]: Better we be specific and call it RangeInt or RangeFloat fields

**Review by pskd73 — CHANGES_REQUESTED** [review]

File: pydictable/field.py (hunk 1 above)
Comment (sattyamjjain) [reply pydictable/field.py]: If i get it from self in Field, it will break for other fields, we need that

Comment (pskd73) [reply pydictable/field.py]: Why it breaks other fields? How is it different from accessing it from self and this function?

**Review by pskd73 — CHANGES_REQUESTED** [review]

File: pydictable/test_core.py [hunk 4]
```diff
@@ -800,3 +801,63 @@ class Number(DictAble):
...
+            self.assertEqual(e.err, 'Pre check failed: 12345 for panNumber should be in proper format')
+
+        profile = Profile(dict={'email': 'abc@gmail.com', 'panNumber': 'ABCDE1234H'})
+
+        self.assertEqual(
+            profile.get_input_spec(),
+            {
+                'email': {
+                    'type': 'RegexField', 'required': False,
+                    'regex': '^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$'
```
Comment (pskd73) [inline pydictable/test_core.py]: This is cool. We get regex also in input spec

**Review by pskd73 — APPROVED** [review]

**Pull request merged**